nixos/modules/services/desktops/pipewire.nix

```diff
@@ -25,17 +35,76 @@ in {
           Automatically run pipewire when connections are made to the pipewire socket.
         '';
       };
+
+      alsa = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Route audio to/from generic ALSA-using applications via the ALSA PIPEWIRE PCM plugin.
+          '';
+        };
+
+        support32Bit = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Whether to support sound for 32-bit ALSA applications on a 64-bit system.
+          '';
+        };
+      };
+
+      jack = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent JACK audio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
+
+      pulse = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent PulseAudio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
     };
   };
 
 
   ###### implementation
   config = mkIf cfg.enable {
-    environment.systemPackages = packages;
+    environment.systemPackages = [ pkgs.pipewire ]
+                                 ++ lib.optional cfg.jack.enable jack-libs
+                                 ++ lib.optional cfg.pulse.enable pulse-libs;
 
-    systemd.packages = packages;
+    systemd.packages = [ pkgs.pipewire ];
 
     systemd.user.sockets.pipewire.wantedBy = lib.mkIf cfg.socketActivation [ "sockets.target" ];
-  };
 
+    # If any paths are updated here they must also be updated in the package test.
+    sound.extraConfig = mkIf cfg.alsa.enable ''
+      pcm_type.pipewire {
+        libs.native = ${pkgs.pipewire.lib}/lib/alsa-lib/libasound_module_pcm_pipewire.so ;
+        ${optionalString enable32BitAlsaPlugins
+          "libs.32Bit = ${pkgs.pkgsi686Linux.pipewire.lib}/lib/alsa-lib/libasound_module_pcm_pipewire.so ;"}
+      }
+      pcm.!default {
+        @func getenv
+        vars [ PCM ]
+        default "plug:pipewire"
+        playback_mode "-1"
+        capture_mode "-1"
+      }
+    '';
+    environment.etc."alsa/conf.d/50-pipewire.conf" = mkIf cfg.alsa.enable {
+      source = "${pkgs.pipewire}/share/alsa/alsa.conf.d/50-pipewire.conf";
```

Comment:
I'm not too sure but in my configuration I included this file with `<${pkgs.pipewire}/share/alsa/alsa.conf.d/50-pipewire.conf>` inside asoundrc.

Reply:
I believe this is the preferred way as it is what the JACK module uses.

https://github.com/NixOS/nixpkgs/blob/dd0a47e7ae2f6f33298c61f75aff2652e0a2f6d6/nixos/modules/services/audio/jack.nix#L237